src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -391,27 +452,42 @@ class TaskExecutor {
   void on_response_recv(rpc::RPCResponse response) {
     std::unique_lock<std::mutex> l(inflight_messages_mutex_);
     auto keys = inflight_messages_[response.first];
+    boost::shared_lock<boost::shared_mutex> metrics_lock(model_metrics_mutex_);
 
     inflight_messages_.erase(response.first);
     std::vector<float> deserialized_outputs =
         deserialize_outputs(response.second);
     assert(deserialized_outputs.size() == keys.size());
     int batch_size = keys.size();
     predictions_counter->increment(batch_size);
```

Comment:
This counter is also incremented when prediction tasks are scheduled via `schedule_predictions`. As a result, in the case where a single application is registered, the aggregate predictions count incorrectly exceeds the application prediction count.